accounts/src/main/java/eu/kevin/accounts/accountsession/AccountSession.kt

```diff
@@ -70,20 +72,58 @@ internal class AccountSession(
     fun beginFlow(listener: AccountSessionListener?) {
         sessionListener = listener
 
-        if (configuration.accountLinkingType == AccountLinkingType.BANK && configuration.preselectedBank != null) {
-            sessionListener?.showLoading(true)
-            lifecycleOwner.lifecycleScope.launch(Dispatchers.IO) {
-                val selectedBank = getSelectedBank()
-                withContext(Dispatchers.Main) {
-                    sessionListener?.showLoading(false)
-                    initializeFlow(selectedBank)
-                }
-            }
+        if (configuration.accountLinkingType == AccountLinkingType.BANK) {
+            validateBanksAndInitializeFlow()
         } else {
             initializeFlow(selectedBank = null)
         }
     }
 
+    private fun validateBanksAndInitializeFlow() {
+        sessionListener?.showLoading(true)
+        lifecycleOwner.lifecycleScope.launch(Dispatchers.IO) {
```

Comment:
The SDK does actually have CoroutineDispatchers class available. I would pass it to the UseCase itself and dispatch to IO inside the UseCase. This way you will clear a lot of withContext hustle here.